File: findDifferenceArray.py
class Solution:
# Approach 1- HashSet
    def findDifference(self, nums1, nums2): # TC O(n+m) // SC O(max(n, m))
        seen=set(nums2)
        answer=[set() for _ in range(2)]
        for num1 in nums1:
            if num1 not in seen:
                answer[0].add(num1)
        
        seen.clear()
        seen=set(nums1)
        for num2 in nums2:
            if num2 not in seen:
                answer[1].add(num2)
        
        return answer

s=Solution()
print(s.findDifference([1,2,3],[2,4,6]))

File: findDifferenceArray.java
import java.util.*;
// Approach 1- HashSet
class Solution { // TC O(n+m) SC O(max(n, m))
    public List<Integer> getElementOfFirstList(int[] nums1, int[] nums2){
        Set<Integer> seen=new HashSet<>();
        for(int num2: nums2){
            seen.add(num2);
        }
        
        Set<Integer> firstList=new HashSet<>();
        for(int num1: nums1){
            if (!seen.contains(num1)){
                firstList.add(num1);
            }
        }
        return new ArrayList<>(firstList);
        
    }
    
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        
        return Arrays.asList(getElementOfFirstList(nums1, nums2), getElementOfFirstList(nums2, nums1));
        
    }
}

class findDifferenceArray{
    public static void main(String[] args) {
        int[] nums1={1,2,3};
        int[] nums2={2,4,6};
        Solution s= new Solution();
        System.out.println(s.findDifference(nums1, nums2));
    }
}

